src/range.h

```diff
@@ -46,11 +52,12 @@
 **  otherwise.  Note that a list for which 'IS_RANGE' returns  0 may still be
 **  a range, but  the kernel does not know  this yet.  Use  'IsRange' to test
 **  whether a list is a range.
-**
-**  Note that  'IS_RANGE' is a  macro, so do not  call it with arguments that
-**  have side effects.
 */
-#define IS_RANGE(val)   (TNUM_OBJ(val)==T_RANGE_NSORT || TNUM_OBJ(val)==T_RANGE_SSORT)
+static inline Int IS_RANGE(Obj val)
+{
+    return TNUM_OBJ(val) >= T_RANGE_NSORT &&
+           TNUM_OBJ(val) <= T_RANGE_SSORT + IMMUTABLE;
+}
```

Comment:
In practice, because no assertions were triggered by not having one. I tried to only add non-standard functions where they were required to make GAP work, and the PLIST functions seem to get used (and abused) alot, so they needed special treatment.